### PR TITLE
feat: complete roll-for-start implementation with comprehensive test coverage

### PR DESCRIPTION
**Pull request by nodots**
## Summary
- Complete roll-for-start functionality with automatic handling and functional programming approach
- Comprehensive test coverage for roll-for-start scenarios including edge cases
- Updated serialization utilities to handle new game states properly
- Fixed various test files to match new roll-for-start behavior patterns
- Added new roll-for-start-fix.test.ts for edge case verification

## Test plan
- [x] All existing tests pass with updated roll-for-start behavior
- [x] New roll-for-start test cases cover edge scenarios
- [x] Serialization tests verify proper state handling
- [x] Game flow tests confirm proper turn transitions
- [x] Bear-off and movement tests work with new game states

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/Board/gnuPositionId.ts [hunk 1]
```diff
@@ -31,13 +31,13 @@ function getPlayerAndOpponent(game: BackgammonGame): {
       // This state is just before the first actual turn where checkers move.
       // GNU Pos ID usually represents a position where someone is about to move checkers.
       // Depending on strict interpretation, one might disallow this state or use this activePlayer.
-      playerOnRoll = (game as any).activePlayer // Assuming activePlayer here means the one to start.
+      playerOnRoll = (game as unknown as BackgammonGameRolled).activePlayer // Assuming activePlayer here means the one to start.
```
Comment (nodots) [inline src/Board/gnuPositionId.ts:34]: This is a bad assumption and casting like this is a code smell. Fix.

**Review by nodots — COMMENTED** [review]
One issue that needs to be resolved.
